model/task/task.go

```diff
@@ -2501,18 +2513,31 @@ func (t *Task) Insert() error {
 // into the old_tasks collection. If this is a display task, its execution tasks
 // are also archived.
 func (t *Task) Archive() error {
+	if !utility.StringSliceContains(evergreen.TaskCompletedStatuses, t.Status) {
+		grip.Debug(message.Fields{
+			"message":   "task is in incomplete state, skipping archiving",
+			"task_id":   t.Id,
+			"execution": t.Execution,
+			"func":      "Archive",
+		})
+		return nil
+	}
 	if t.DisplayOnly && len(t.ExecutionTasks) > 0 {
 		return errors.Wrapf(ArchiveMany([]Task{*t}), "archiving display task '%s'", t.Id)
 	} else {
 		// Archiving a single task.
 		archiveTask := t.makeArchivedTask()
 		err := db.Insert(OldCollection, archiveTask)
-		if err != nil {
+		if err != nil && !db.IsDuplicateKey(err) {
 			return errors.Wrap(err, "inserting archived task into old tasks")
 		}
 		t.Aborted = false
 		err = UpdateOne(
-			bson.M{IdKey: t.Id},
+			bson.M{
+				IdKey:       t.Id,
+				StatusKey:   bson.M{"$in": evergreen.TaskCompletedStatuses},
+				CanResetKey: false,
+			},
```

Comment:
For `UpdateOne`, Anser will [return a not found error if no docs were modified](https://github.com/mongodb/anser/blob/99dd61768f4a246b31e8e3ff8b2d0e51760f2194/db/wrapper.go#L207-L209), so we would have to return nil error for `adb.ResultsNotFound` (i.e. the task has already been archived) just in case the task has already been archived.

Reply:
Ah, thank you